### PR TITLE
Extend --help with all current command line args

### PR DESCRIPTION
**Pull request by chamons**
- Fixes https://github.com/Azure/autorest.go/issues/663

```
Go Generator (activated by --go)

  --module=<string>             The name of the Go module written to go.mod.  Omit to skip go.mod generation.
  --file-prefix=<string>        Optional prefix to file names. For example, if you set your file prefix to "zzz_", all generated code files will begin with "zzz_".
  --export-clients=<boolean>    Indicates if generated clients are to be exported.  Default to true for ARM, false for data-plane.
  --header-text=<string>        Text to include as a header comment in generated files (magic strings:MICROSOFT_MIT, MICROSOFT_APACHE, MICROSOFT_MIT_NO_VERSION, MICROSOFT_APACHE_NO_VERSION, MICROSOFT_MIT_NO_CODEGEN)
  --output-folder=<string>      Target folder for generated artifacts.
  --openapi-type=<string>       Open API Type: 'arm' or 'data-plane'.
  --azure-arm=<boolean>         Generate code in Azure flavor.
  --head-as-boolean=<boolean>   With this flag, HEAD calls to non-existent resources (404) will not raise an error. Instead, if the resource exists, we return true, else false. Forced to be true if --azure-arm is set, otherwise defaults to false.
```

### CODE REVIEW

**Comment by chamons** [comment]
Text mostly taken from files in autorest core with similar or identical names.

File: src/autorest-configuration.md [hunk 1]
```diff
@@ -68,4 +68,19 @@ help-content:
       - key: export-clients
         type: boolean
         description: Indicates if generated clients are to be exported.  Default to true for ARM, false for data-plane.
+      - key: header-text
```
Comment (jhendrixMSFT) [inline src/autorest-configuration.md:71]: Shouldn't these be added to the autorest front-end?  They're not specific to autorest.go

Comment (chamons) [reply src/autorest-configuration.md:71]: Thoughts @timotheeguerin ?

Comment (chamons) [reply src/autorest-configuration.md:71]: I slapped `autorest --go --help` and did not see them.

Comment (timotheeguerin) [reply src/autorest-configuration.md:71]: Yeah I don't think the autorest help content contains all the options.

It is here if you want to add missing ones. https://github.com/Azure/autorest/blob/main/packages/libs/configuration/resources/help-configuration.md

**Comment by chlowell** [comment]
Closed in favor of https://github.com/Azure/autorest/pull/4299

**Pull request closed without merging**